From 1a1fa2414443ce07b452a14d1415839152c13120 Mon Sep 17 00:00:00 2001
From: Khiem Nguyen <khiem.nguyen.xt@renesas.com>
Date: Wed, 20 Mar 2019 12:59:35 +0700
Subject: [PATCH 757/909] mfd: bd9571mwv: Enable GPIO support for PMIC
 BD9574MWF

commit b0375b0bb391fabd22b4bd81df7a5aa18055f21a from
git://git.kernel.org/pub/scm/linux/kernel/git/horms/renesas-bsp.git

The BD9574MWF supports small PMIC GPIO block.
This patch enables GPIO control for that block.

Signed-off-by: Khiem Nguyen <khiem.nguyen.xt@renesas.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/mfd/bd9571mwv.c | 18 +++++++++++++++---
 1 file changed, 15 insertions(+), 3 deletions(-)

diff --git a/drivers/mfd/bd9571mwv.c b/drivers/mfd/bd9571mwv.c
index 657f40f61391..bb0f4ae68cc1 100644
--- a/drivers/mfd/bd9571mwv.c
+++ b/drivers/mfd/bd9571mwv.c
@@ -29,6 +29,10 @@ static const struct mfd_cell bd9571mwv_cells[] = {
 	{ .name = "bd9571mwv-gpio", },
 };
 
+static const struct mfd_cell bd9574mwf_cells[] = {
+	{ .name = "bd9571mwv-gpio", },
+};
+
 static const struct bd957x_data *bd_data;
 
 /* Regmap for BD9571MWV */
@@ -315,9 +319,17 @@ static int bd9571mwv_probe(struct i2c_client *client,
 		return ret;
 	}
 
-	ret = mfd_add_devices(bd->dev, PLATFORM_DEVID_AUTO, bd9571mwv_cells,
-			      ARRAY_SIZE(bd9571mwv_cells), NULL, 0,
-			      regmap_irq_get_domain(bd->irq_data));
+	if (product_code == BD9571MWV_PRODUCT_CODE_VAL) {
+		ret = mfd_add_devices(bd->dev, PLATFORM_DEVID_AUTO,
+				      bd9571mwv_cells,
+				      ARRAY_SIZE(bd9571mwv_cells), NULL, 0,
+				      regmap_irq_get_domain(bd->irq_data));
+	} else {
+		ret = mfd_add_devices(bd->dev, PLATFORM_DEVID_AUTO,
+				      bd9574mwf_cells,
+				      ARRAY_SIZE(bd9574mwf_cells), NULL, 0,
+				      regmap_irq_get_domain(bd->irq_data));
+	}
 	if (ret) {
 		regmap_del_irq_chip(bd->irq, bd->irq_data);
 		return ret;
-- 
2.17.1

